src/timer/TimerProvider.jsx

```diff
@@ -52,15 +49,19 @@ const TimerProvider = ({
     },
   ).join(':');
 
-  const pollExam = () => {
+  const pollExam = useCallback(() => {
     // poll url may be null if this is an LTI exam
-    dispatch(pollAttempt(attempt.exam_started_poll_url));
-  };
+    if (attempt?.exam_started_poll_url) {
+      dispatch(pollAttempt(attempt.exam_started_poll_url));
+    }
+  }, [attempt.exam_started_poll_url, dispatch]);
+  const processTimeLeft = useCallback(() => (secondsLeft) => {
+    const criticalLowTime = timeLimitMins * 60 * TIME_LIMIT_CRITICAL_PCT;
+    const lowTime = timeLimitMins * 60 * TIME_LIMIT_LOW_PCT;
```

Comment:
These constants were renamed and relocated since they are not actual app constants, but just calculations based on context. The uppercase naming was misleading.